mempool/txpool/legacypool/legacypool.go

```diff
@@ -1402,12 +1413,37 @@ func (pool *LegacyPool) promoteExecutables(accounts []common.Address) []*types.T
 		}
 		log.Trace("Removed old queued transactions", "count", len(forwards))
 		// Drop all transactions that are too costly (low balance or out of gas)
-		drops, _ := list.Filter(pool.currentState.GetBalance(addr), gasLimit)
-		for _, tx := range drops {
+		costDrops, _ := list.CostFilter(pool.currentState.GetBalance(addr), gasLimit)
```

Comment:
I wonder if the separate cost filter is still needed since we technically have one in the antehandler via a combination of https://github.com/cosmos/evm/blob/befde4fafa15c469a181708043224c95dea66c52/ante/evm/07_can_transfer.go, https://github.com/cosmos/evm/blob/61eedfcf340e8140b82e9e02478b086b81f63eda/ante/evm/08_gas_consume.go, and https://github.com/cosmos/evm/blob/491aa78f515250590ab235a865233418f387c288/ante/evm/fee_checker.go